### PR TITLE
GH-1887 Add support for DDL Field Type Dataset

### PR DESCRIPTION
**Pull request by GordonSmith**
Fixes GH-1887-DDLFieldTypeDataset

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage @mlzummo @jchambers-ln Please Review

**Comment by jbrundage** [comment]
@GordonSmith This looks good, but why create an instance of a Table? Could you give a practical use case for this dataset fieldType to help me understand?

File: src/marshaller/HipieDDL.js [hunk 1]
```diff
@@ -239,6 +245,35 @@
...
+                                    ;
+                                    row[cellIdx] = table;
+                                }
+                            });
+                            return row;
+                        });
+                        break;
                     case "widget":
                         retVal = retVal.map(function (row) {
                             //  TODO:  Insert Widget
```
Comment (jchambers-ln) [inline src/marshaller/HipieDDL.js]: existing todo:

**Comment by jchambers-ln** [comment]
one todo still here, other than that looks good

**Comment by GordonSmith** [comment]
@jbrundage  This is the "quick" and "general" solution to arrays of data that happen to contain arrays of data within any given cell (child datasets) - there is another PR which allows the user to specify specific widget types with mappings etc.

**Comment by GordonSmith** [comment]
@jchambers-ln That todo is resolved in another PR.

**Pull request merged**